### ISSUE TITLE
Mongoose buffering when logging in.

### ISSUE THREAD

**Issue by cheng-alvin** [Chillandchat/api#3]
### What happened 
A Chill&chat API bug has been logged at: ```15/1/2023 0:26:0 UTC``` or ```1673702760```(UNIX time UTC) on the production server. This will affect all Chill&chat login and database query attempts made from that time stamp onwards.  The API produced the following logs regarding the crash which is listed below:
```
ERROR: [15/1/2023 0:26:0] MongooseError: Operation `users.findOne()` buffering timed out after 10000ms
(node:196) UnhandledPromiseRejectionWarning: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://docs.atlas.mongodb.com/security-whitelist/
    at NativeConnection.Connection.openUri (/home/runner/api/node_modules/mongoose/lib/connection.js:825:32)
    at /home/runner/api/node_modules/mongoose/lib/index.js:409:10
    at /home/runner/api/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
    at new Promise (<anonymous>)
    at promiseOrCallback (/home/runner/api/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
    at Mongoose._promiseOrCallback (/home/runner/api/node_modules/mongoose/lib/index.js:1262:10)
    at Mongoose.connect (/home/runner/api/node_modules/mongoose/lib/index.js:408:20)
    at Object.<anonymous> (/home/runner/api/src/index.ts:79:10)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Module.m._compile (/home/runner/api/node_modules/ts-node/src/index.ts:1618:23)
(node:196) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:196) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

### Reproduce
API version: v1.8.4 (latest as of 15/1/2023)
OS: Linux based
Runtime: ```ts-node```

Steps:
- Spin up environment as listed above
- Send POST request to ```/api/login```.

**Comment by cheng-alvin** [comment]
Local machine test results:
I have tested the API at ```1a9a6c659f0f870736ddc860e91181a49e4c421d``` and it seems to work fine when logging in.

**Comment by cheng-alvin** [comment]
LOGS:
```
[15/1/2023 11:48:12] Debugger tools intialized.
(node:45693) [MONGOOSE] DeprecationWarning: Mongoose: the `strictQuery` option will be switched back to `false` by default in Mongoose 7. Use `mongoose.set('strictQuery', false);` if you want to prepare for this change. Or use `mongoose.set('strictQuery', true);` to suppress this warning.
(Use `node --trace-deprecation ...` to show where the warning was created)
[15/1/2023 11:48:12] Server listening on port 3000
[15/1/2023 11:49:16] Alvin login success
```

**Comment by cheng-alvin** [comment]
As the logs mentioned, it says that I need to whitelist the IP address. However, the address is already set to ```0.0.0.0/0``` which means all IPs

**Comment by cheng-alvin** [comment]
It seems to work now after a restart of the API, and all the other functions seems to return back to normal working order.

**Comment by cheng-alvin** [comment]
As mentioned in https://mongoosejs.com/docs/connections.html#error-handling
, the ```mongoose.connect()``` function is asynchronous. So you have to wrap the function in a async and await function. I am guessing that it could be cause by the asynchronous nature. Maybe we should wait for the server to connect before running our command.

**Comment by cheng-alvin** [comment]
This issue has been merged and reviewed by @cheng-alvin in #4
